### PR TITLE
Lkt lowering: enhance big integers handling

### PR DESCRIPTION
**Pull request by pmderodat**

### CODE REVIEW

File: testsuite/tests/contrib/lkt_semantic/array_type_concrete/test.out [hunk 1]
```diff
@@ -407,19 +407,16 @@ Expr <StringLit test.lkt:31:29-31:32>
...
 
-test.lkt:31:35: error: Mismatched types: expected `Array[String]`, got an array
+test.lkt:31:35: error: Mismatched types: expected `Array[String]`, got `Array[Int]`
 31 | @invalid val a11 = [[bar(), "2"], [1, 2]]
    |                                   ^^^^^^
 
-test.lkt:31:20: error: ambiguous type for expression
-31 | @invalid val a11 = [[bar(), "2"], [1, 2]]
-   |                    ^^^^^^^^^^^^^^^^^^^^^^
+Expr <ArrayLiteral test.lkt:31:20-31:42>
```
Comment (thvnx) [inline testsuite/tests/contrib/lkt_semantic/array_type_concrete/test.out:420]: Shouldn't still be ambiguous here?

File: contrib/lkt/language/parser.py [hunk 2]
```diff
@@ -129,6 +129,14 @@ class ParamMatch(Struct):
     formal = UserField(type=T.FormalParam, default_value=No(T.FormalParam))
```
Comment (thvnx) [inline contrib/lkt/language/parser.py:129]: Comment about the commit itself. I guess it shouldn't be in this PR?

File: testsuite/tests/contrib/lkt_semantic/array_type_concrete/test.out (hunk 1 above)
Comment (pmderodat) [reply testsuite/tests/contrib/lkt_semantic/array_type_concrete/test.out:420]: Why should it be? Note that we still have one error:
```
test.lkt:31:35: error: Mismatched types: expected `Array[String]`, got `Array[Int]`
31 | @invalid val a11 = [[bar(), "2"], [1, 2]]
   |                                   ^^^^^^
```
so in a way we’re doing a better error reporting job, now (less noise, still has the useful bit).

File: contrib/lkt/language/parser.py (hunk 2 above)
Comment (pmderodat) [reply contrib/lkt/language/parser.py:129]: I suppose you are referring to “Lkt: complete typing of "if" expressions” (27eb3ce). Yes indeed it should not be there, I’ll remove it, thank you!

File: testsuite/tests/contrib/lkt_semantic/array_type_concrete/test.out (hunk 1 above)
Comment (thvnx) [reply testsuite/tests/contrib/lkt_semantic/array_type_concrete/test.out:420]: My bad, didn't see that!

**Review by thvnx — APPROVED** [review]
LGTM

**Review by raph-amiard — APPROVED** [review]

**Pull request merged**